sim/make/build.mk

```diff
@@ -0,0 +1,87 @@
+# See LICENSE for license details.
+
+################################################################################
+# Helper to run SBT in the project.
+################################################################################
+
+SBT_COMMAND ?= shell
+SBT_NON_THIN ?= $(subst $(SBT_CLIENT_FLAG),,$(SBT))
+.PHONY: sbt
+sbt:
+	cd $(base_dir) && $(SBT_NON_THIN) ";project $(firesim_sbt_project); $(SBT_COMMAND)"
+
+
+################################################################################
+# Target Configuration
+################################################################################
+
+firesim_source_dirs = \
+	$(addprefix $(firesim_base_dir)/,\
+			. \
+			midas \
+			midas/targetutils \
+			firesim-lib\
+	)
+firesim_main_srcs = $(foreach submodule, $(firesim_source_dirs),\
+	$(shell find $(submodule)/ -iname "[!.]*.scala" -print 2> /dev/null | grep -E 'src/main/scala'))
+firesim_test_srcs = $(foreach submodule, $(firesim_source_dirs),\
+	$(shell find $(submodule)/ -iname "[!.]*.scala" -print 2> /dev/null | grep -E 'src/test/scala'))
+
+TARGET_SBT_PROJECT ?= $(firesim_sbt_project)
+
+TARGET_SOURCE_DIRS ?= \
+		$(addprefix $(chipyard_dir)/generators/, rocket-chip/src, rocket-chip/api-config-chipsalliance)
+
+TARGET_SRCS = $(foreach submodule,$(TARGET_SOURCE_DIRS),\
+	$(shell find $(submodule)/ -iname "[!.]*.scala" -print 2> /dev/null | grep 'src/main/scala'))
+
+################################################################################
+# Build jars using SBT assembly and cache them.
+################################################################################
+
+define build_classpath
+	$(SBT_NON_THIN) \
+		"set showSuccess := false; project $(1); compile; package"
+	$(SBT_NON_THIN) \
+		--error \
+		"set showSuccess := false; project $(1); compile; package; export $(2):fullClasspath" \
+		| head -n 1 | tr -d '\n' > $@
```

Comment:
Does this have to be done without the thin client?

Reply:
I don't know. But this is what's been used before, so it's not changing here.